showcase/src/app/shared/example-viewer/example-viewer.spec.ts

```diff
@@ -0,0 +1,92 @@
+import { HttpClientModule } from "@angular/common/http";
+import { BrowserAnimationsModule } from "@angular/platform-browser/animations";
+import { async, ComponentFixtureAutoDetect, ComponentFixture, TestBed } from "@angular/core/testing";
+import { MatButtonModule, MatIconModule, MatTabsModule, MatTooltipModule } from "@angular/material";
+import * as hljs from "highlight.js";
+import { HighlightJsModule, HIGHLIGHT_JS } from "angular-highlight-js";
+
+import { ExampleViewerComponent } from "./example-viewer.component";
+import { ExampleViewerService } from "./example-viewer.service";
+import { throwError, of } from "rxjs";
+
+export function highlightJsFactory(): any {
+	return hljs;
+}
+fdescribe("ExampleViewerComponent", () => {
+	let component: ExampleViewerComponent;
+	let exampleViewerService: ExampleViewerService;
+	let fixture: ComponentFixture<ExampleViewerComponent>;
+
+	beforeEach(async(() => {
+		return TestBed.configureTestingModule({
+			declarations: [ExampleViewerComponent],
+			imports: [
+				BrowserAnimationsModule,
+				HttpClientModule,
+				HighlightJsModule.forRoot({
+					provide: HIGHLIGHT_JS,
+					useFactory: highlightJsFactory
+				}),
+				MatButtonModule,
+				MatIconModule,
+				MatTabsModule,
+				MatTooltipModule
+			],
+			providers: [{ provide: ComponentFixtureAutoDetect, useValue: true }, ExampleViewerService]
+		}).compileComponents();
+	}));
+
+	beforeEach(() => {
+		fixture = TestBed.createComponent(ExampleViewerComponent);
+		component = fixture.componentInstance;
+		exampleViewerService = fixture.debugElement.injector.get<ExampleViewerService>(ExampleViewerService as any);
+	});
+
+	describe("@Input() title", () => {
+		it("should change the title according to the @Input", () => {
+			const h2: any = fixture.nativeElement.querySelector(".example-viewer-title-spacer h2");
+			component.title = "Test title";
+			fixture.detectChanges();
+			expect(h2.textContent).toContain(component.title);
+		});
+	});
+
+	describe("example-viewer-service", () => {
```

Comment:
This test suite ("describe" block) does not belong here. This spec.ts file are only related to the example viewer component.

The tests related to the service should be in a different spec.ts file ;)